WalletWasabi/WabiSabi/Client/InternalDestinationProvider.cs

```diff
@@ -18,17 +18,15 @@ public IEnumerable<IDestination> GetNextDestinations(int count, bool preferTapro
 	{
 		// Get all locked internal keys we have and assert we have enough.
 		KeyManager.AssertLockedInternalKeysIndexedAndPersist(count, preferTaproot);
-		var preferedScriptPubKeyType = preferTaproot
-			? ScriptPubKeyType.TaprootBIP86
-			: ScriptPubKeyType.Segwit;
 
-		bool IsAvailable(HdPubKey hdPubKey) =>
-			hdPubKey.IsInternal &&
-			hdPubKey.KeyState == KeyState.Locked &&
-			hdPubKey.FullKeyPath.GetScriptTypeFromKeyPath() == preferedScriptPubKeyType;
-
-		return KeyManager
-			.GetKeys(IsAvailable)
-			.Select(x => x.GetAddress(KeyManager.GetNetwork()));
+		var allKeys = KeyManager.GetNextCoinJoinKeys().ToList();
+		var taprootKeys = allKeys
+			.Where(x => x.FullKeyPath.GetScriptTypeFromKeyPath() == ScriptPubKeyType.TaprootBIP86)
+			.ToList();
+		
+		var destinations = preferTaproot && taprootKeys.Count >= count
```

Comment:
So if there are not enough keys locked in the KeyManager (for any reason), this will take `allKeys` that can contain mixed script types? Shouldn't this fail hard if the desired count from the preferred script time is missing?

Reply:
I don't think so. A preference is not a requirement.